lib/location/location_core.c

```diff
@@ -383,6 +383,13 @@ static int location_core_location_get_pos(void)
 
 	err = location_method_api_get(requested_method)->location_get(&loc_req_info);
 
+	if (IS_ENABLED(CONFIG_LOCATION_DATA_DETAILS) && err == 0) {
+		struct location_event_data request_started;
+
+		request_started.id = LOCATION_EVT_STARTED;
```

Comment:
It would be good to zero the rest of the structure, looks like that's already done for other events.

I also started wondering if the method should be set in the event.
```suggestion
		struct location_event_data request_started = {
			.id = LOCATION_EVT_STARTED
		};
```